### PR TITLE
Convert EventEmitter to TypeScript and add tests

### PR DESCRIPTION
**Pull request by Tyriar**
Part of #335

### CODE REVIEW

File: src/EventEmitter.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+};
+
+export class EventEmitter {
+  private _events: {[type: string]: ListenerType[]};
+
+  constructor() {
+    this._events = this._events || {};
+  }
+
+  // TODO: Merge addListener and on, no reason for an alias in a private component
```
Comment (parisk) [inline src/EventEmitter.ts]: Should we merge `addListener` and `on` before merging?

File: src/EventEmitter.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    this._events[type] = this._events[type] || [];
+    this._events[type].push(listener);
+  }
+
+  public on(type, listener): void {
+    this.addListener(type, listener);
+  }
+
+  // TODO: Merge removeListener and off, no reason for an alias in a private component
+  public removeListener(type, listener): void {
```
Comment (parisk) [inline src/EventEmitter.ts]: Should we merge `removeListener` and `off` before merging?

File: src/EventEmitter.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+interface ListenerType {
+    (): void;
+    listener?: () => void;
+};
+
+export class EventEmitter {
+  private _events: {[type: string]: ListenerType[]};
+
+  constructor() {
+    this._events = this._events || {};
```
Comment (Tyriar) [inline src/EventEmitter.ts]: https://github.com/sourcelair/xterm.js/pull/432/commits/937cf43e091f8752515d83a99ad7bd91f1389c08

File: src/EventEmitter.ts (hunk 1 above)
Comment (Tyriar) [inline src/EventEmitter.ts]: https://github.com/sourcelair/xterm.js/pull/432/commits/0d9d3aca1a2d306f456ef95d2e33c7941a231c3a

File: src/EventEmitter.ts (hunk 2 above)
Comment (Tyriar) [inline src/EventEmitter.ts]: https://github.com/sourcelair/xterm.js/pull/432/commits/0d9d3aca1a2d306f456ef95d2e33c7941a231c3a https://github.com/sourcelair/xterm.js/pull/432/commits/965ebde22f6ad8f7ea4866a5133cd78302918203

**Pull request merged**